server.js

```diff
@@ -0,0 +1,66 @@
+import express from "express";
+import cors from 'cors'
+const app = express();
+let PORT = 5000
+
+import { join, dirname } from "path";
+import { Low, JSONFile } from "lowdb";
+import { fileURLToPath } from "url";
+import { v4 as uuidv4 } from "uuid";
+
+const __dirname = dirname(fileURLToPath(import.meta.url));
+const file = join(__dirname, "data/db.json");
+const adapter = new JSONFile(file);
+const db = new Low(adapter);
+await db.read();
+db.data = db.data || { users: [] }
+
+app.use( express.json() )
+app.use( cors() )
+
+
+app.post('/signup', async (req, res) => {
+    // console.log(req.body.name)
+    let user = {
+        _id: uuidv4(),
+        username: req.body.username,
+        password: req.body.password
+    }
+    const { username } = req.body
+
+    const existingUsername = db.data.users.find(user => user.username === username)
+
+    if(existingUsername){
+        return res.json({error: "already existing username"})
+    }
+    
+    db.data.users.push(user)
+    await db.write()
+    res.json(user)
+})
+
+
+app.post('/login', (req, res)=>{
+    const { username } = req.body
+    const { password } = req.body
+    const corrName = db.data.users.find(user => user.username === username)
+    const corrPw = db.data.users.find(user => user.password === password)
+    
+    console.log(corrName)
+
+    if(!corrName || !corrPw){
```

Comment:
That check here unfortunately is not enough :)
Because  you checked if any user with a username exists and any user with such a password exist, but not if one user with BOTH exists..
E.g. if we have users "rob" and "vasilis" in your DB. Rob has password R123 and Vasilis has password V123
If I provide now the user "rob" but with vasilis' password "V123" your code would log me in, even though that is not my user password.

So even though I like that you do not make it easier for me to login, it might become a security issue :)

Reply:
I noticed....aaaand forgot to fix it haha